src/libAtomVM/otp_net.c

```diff
@@ -325,6 +326,38 @@ static term nif_net_getaddrinfo(Context *ctx, int argc, term argv[])
     return ret;
 }
 
+//
+// net:gethostname/0
+//
+#ifdef HAVE_GETHOSTNAME
+static term nif_net_gethostname(Context *ctx, int argc, term argv[])
+{
+    TRACE("nif_net_gethostname\n");
+    UNUSED(argc);
+    UNUSED(argv);
+
+    char buf[256];
+    int r = gethostname(buf, sizeof(buf));
+    if (UNLIKELY(r)) {
+        if (UNLIKELY(memory_ensure_free_opt(ctx, TUPLE_SIZE(2), MEMORY_CAN_SHRINK) != MEMORY_GC_OK)) {
+            RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+        }
+        return make_error_tuple(posix_errno_to_term(errno, ctx->global), ctx);
+    }
+
+    size_t len = strlen(buf);
+    if (UNLIKELY(memory_ensure_free_opt(ctx, TUPLE_SIZE(2) + len * CONS_SIZE, MEMORY_CAN_SHRINK) != MEMORY_GC_OK)) {
```

Comment:
What about: `LIST_SIZE(len, 1)` ? It is the same of `len * CONS_SIZE` but maybe more readable.